### PR TITLE
Migration5

### PR DESCRIPTION
**Pull request by ElizabethSamuel-MSFT**
Review AutoSave and Co-authoring articles

### CODE REVIEW

**Comment by saldana** [comment]
### :x: Validation status: errors


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[]() | :x:Error | | [Details](#user-content-)

<a name=""></a>
### []()
 - **[Error]** Cannot sync git repo to specified state. Please make sure the pull request is mergeable. Details: Git merge results in conflict, repository remote url: https://github.com/OfficeDev/VBA-content, working directory: W:\h1nw-s, target branch: master, target branch commit: d855b07a194571c5d763c859430de03279d5d396, source branch: Migration5, source branch commit: .

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C6%5C6%5Cf8b5b3f0-7390-2e9e-6625-a74fa1ae6f0a%5CPullRequest%5C201706061846580844-322%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by saldana** [comment]
### :x: Validation status: errors


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[]() | :x:Error | | [Details](#user-content-)

<a name=""></a>
### []()
 - **[Error]** Cannot sync git repo to specified state. Please make sure the pull request is mergeable. Details: Git merge results in conflict, repository remote url: https://github.com/OfficeDev/VBA-content, working directory: W:\qgkv-s, target branch: master, target branch commit: d855b07a194571c5d763c859430de03279d5d396, source branch: Migration5, source branch commit: .

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C6%5C6%5Cf8b5b3f0-7390-2e9e-6625-a74fa1ae6f0a%5CPullRequest%5C201706061847073179-322%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by saldana** [comment]
### :x: Validation status: errors


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[]() | :x:Error | | [Details](#user-content-)

<a name=""></a>
### []()
 - **[Error]** Cannot sync git repo to specified state. Please make sure the pull request is mergeable. Details: Git merge results in conflict, repository remote url: https://github.com/OfficeDev/VBA-content, working directory: W:\z0zq-s, target branch: master, target branch commit: d855b07a194571c5d763c859430de03279d5d396, source branch: Migration5, source branch commit: .

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C6%5C6%5Cf8b5b3f0-7390-2e9e-6625-a74fa1ae6f0a%5CPullRequest%5C201706062041175697-322%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by saldana** [comment]
### :x: Validation status: errors


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[]() | :x:Error | | [Details](#user-content-)

<a name=""></a>
### []()
 - **[Error]** Cannot sync git repo to specified state. Please make sure the pull request is mergeable. Details: Git merge results in conflict, repository remote url: https://github.com/OfficeDev/VBA-content, working directory: W:\jalp-s, target branch: master, target branch commit: d855b07a194571c5d763c859430de03279d5d396, source branch: Migration5, source branch commit: .

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C6%5C6%5Cf8b5b3f0-7390-2e9e-6625-a74fa1ae6f0a%5CPullRequest%5C201706062041351366-322%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by saldana** [comment]
### :x: Validation status: errors


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[]() | :x:Error | | [Details](#user-content-)

<a name=""></a>
### []()
 - **[Error]** Cannot sync git repo to specified state. Please make sure the pull request is mergeable. Details: Git merge results in conflict, repository remote url: https://github.com/OfficeDev/VBA-content, working directory: W:\dyov-s, target branch: master, target branch commit: d855b07a194571c5d763c859430de03279d5d396, source branch: Migration5, source branch commit: .

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C6%5C6%5Cf8b5b3f0-7390-2e9e-6625-a74fa1ae6f0a%5CPullRequest%5C201706062213452985-322%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by saldana** [comment]
### :x: Validation status: errors


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[]() | :x:Error | | [Details](#user-content-)

<a name=""></a>
### []()
 - **[Error]** Cannot sync git repo to specified state. Please make sure the pull request is mergeable. Details: Git merge results in conflict, repository remote url: https://github.com/OfficeDev/VBA-content, working directory: W:\l3a1-s, target branch: master, target branch commit: d855b07a194571c5d763c859430de03279d5d396, source branch: Migration5, source branch commit: .

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C6%5C6%5Cf8b5b3f0-7390-2e9e-6625-a74fa1ae6f0a%5CPullRequest%5C201706062213549814-322%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by saldana** [comment]
### :x: Validation status: errors


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[]() | :x:Error | | [Details](#user-content-)

<a name=""></a>
### []()
 - **[Error]** Cannot sync git repo to specified state. Please make sure the pull request is mergeable. Details: Git merge results in conflict, repository remote url: https://github.com/OfficeDev/VBA-content, working directory: W:\qjek-s, target branch: master, target branch commit: d855b07a194571c5d763c859430de03279d5d396, source branch: Migration5, source branch commit: .

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C6%5C6%5Cf8b5b3f0-7390-2e9e-6625-a74fa1ae6f0a%5CPullRequest%5C201706062234591552-322%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by saldana** [comment]
### :x: Validation status: errors


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[]() | :x:Error | | [Details](#user-content-)

<a name=""></a>
### []()
 - **[Error]** Cannot sync git repo to specified state. Please make sure the pull request is mergeable. Details: Git merge results in conflict, repository remote url: https://github.com/OfficeDev/VBA-content, working directory: W:\dkkg-s, target branch: master, target branch commit: d855b07a194571c5d763c859430de03279d5d396, source branch: Migration5, source branch commit: .

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C6%5C6%5Cf8b5b3f0-7390-2e9e-6625-a74fa1ae6f0a%5CPullRequest%5C201706062234492782-322%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: VBA/Excel-VBA/articles/application-workbookafterremotechange-event-excel.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+## Return value
+
+Nothing
+
+## Example
+
+This example shows you where you can place code that runs after merging an incoming remote change. This code must be placed in a class module and an instance of that class must be correctly initialized. For more information about how to use event procedures with the  **Application** object, see [Using Events with the Application Object](http://msdn.microsoft.com/library/0063feba-47fd-29be-d2d5-8fcf47e70cbc%28Office.15%29.aspx).
+
+```vb
```
Comment (astegmaier) [inline VBA/Excel-VBA/articles/application-workbookafterremotechange-event-excel.md]: Good sample code! Just to confirm, did you test it?

File: VBA/Office-Shared-VBA/articles/about-autosave.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
+---
+title: About AutoSave in Office
+ms.prod: MULTIPLEPRODUCTS
+ms.assetid: about-autosave
+---
+
+
+# About AutoSave in Office
+
+Learn about how AutoSave works in Excel, PowerPoint, and Word and how it can impact add-in or macro integration. For more about how AutoSave works in general, see [this link](https://support.office.com/en-US/article/What-is-AutoSave-6d6bd723-ebfd-4e40-b5f6-ae6e8088f7a5).
```
Comment (astegmaier) [inline VBA/Office-Shared-VBA/articles/about-autosave.md]: I would just say "add-ins or macros" rather than "add-in or macro integration"

File: VBA/Office-Shared-VBA/articles/about-autosave.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
+---
+title: About AutoSave in Office
+ms.prod: MULTIPLEPRODUCTS
+ms.assetid: about-autosave
+---
+
+
+# About AutoSave in Office
```
Comment (astegmaier) [inline VBA/Office-Shared-VBA/articles/about-autosave.md]: I think the title should make it clear that this article is about AutoSave + Add-ins/Macros, not AutoSave in general. Something like "AutoSave and Macros and Add-Ins in Office" or "How AutoSave impacts macros and add-ins"

File: VBA/Office-Shared-VBA/articles/about-autosave.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+Currently, Excel, Word, and PowerPoint provide a **BeforeSave** event that allows a developer to execute code after the user triggers a save but before the save actually occurs. Excel also provides an **AfterSave** event that can execute macro or add-in code after the save completes.
+
+When AutoSave is enabled, these events will fire automatically on a periodic basis without user interaction. Because of this new behavior when AutoSave is enabled, add-ins and macros that leverage these events may experience problems when AutoSave is on.
+
+In general, these issues can be avoided if the user chooses to disable AutoSave. You can also do this on the user’s behalf using the **autoSaveOn** property (see the example below). You can also take steps as a developer to mitigate these potential problems so that your add-ins and macros work smoothly, even if AutoSave is enabled.
```
Comment (astegmaier) [inline VBA/Office-Shared-VBA/articles/about-autosave.md]: Make "AutoSaveOn" text a link to the detailed article.

File: VBA/Office-Shared-VBA/articles/about-autosave.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+#### Example Scenario
+
+Imagine an add-in that allows the user to create custom maps based on data in an Excel workbook. Such an add-in might have **BeforeSave** code that serializes any maps that the user has created and stores them in the workbook in a CustomXML part. This process might take a second to complete and Excel is unresponsive while this is happening. When AutoSave is off, the user gets to choose when he or she wants to save and therefore, even though the add-in slows down the save process slightly, the user does not notice. When AutoSave is enabled, this **BeforeSave** code is executed automatically on a periodic basis even if the user is in the middle of something else (like typing data into a cell) which could be extremely annoying.
```
Comment (astegmaier) [inline VBA/Office-Shared-VBA/articles/about-autosave.md]: "and Excel is unresponsive while this is happening -> "and Excel **would be** responsive while this is happening"

File: VBA/Office-Shared-VBA/articles/about-autosave.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+- Consider removing any code that needs to display UI to other areas of the application. For example, there could be a "validate" button that the user could click to trigger the validation process. Or you could fire the validation code only if the user attempts to change the existing data.
+
+- If you want validation code to trigger only on the first save from a new document but not on subsequent auto-saves, consider inspecting a property like Excel's **Workbook.Path** before displaying any UI during **BeforeSave** or **AfterSave**. In Excel, the **Workbook.Path** property should be blank if the workbook does not yet have a save location.
+
+### Issue 3: Code in save events causes the undo stack to get blown away (Excel Only)
+
+#### Description
+
+In general, executing certain VBA statements in Excel will clear the undo stack. For example, changing the value of a cell by executing
+`ActiveCell.Value = "asdf"` will clear the undo stack. If such code is present in the **BeforeSave** or **AfterSave** event for a particular macro or add-in and AutoSave is on, a user of that macro or add-in will frequently not be able to undo normal user actions as expected.
```
Comment (astegmaier) [inline VBA/Office-Shared-VBA/articles/about-autosave.md]: Maybe change "asdf" to "myValue" - "asdf" makes it seemed like a draft/rushed.

File: VBA/Office-Shared-VBA/articles/about-autosave.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+
+#### Example Scenario
+
+An add-in might completely override the default Word save code so that the file is actually saved to a corporate database instead of a disk or sharepoint location. Such an add-in would first cancel the attempted save before trying to save in another place.
+
+#### Workarounds
+
+- Such add-ins should ensure that AutoSave is turned off by setting autoSaveOn to False. Since a file must already be saved in a OneDrive or SharePoint location for AutoSave to be on, AutoSave should already be off in most cases in the scenario above.
+
+## Example
```
Comment (astegmaier) [inline VBA/Office-Shared-VBA/articles/about-autosave.md]: This example seems unrelated to the above content. Should it be moved to the beginning to be with the general sentence about "turning off autosave" being a general workaround to all these problems.

File: VBA/Office-Shared-VBA/articles/about-autosave.md [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+
+Currently, Excel, Word, and PowerPoint provide a **BeforeSave** event that allows a developer to execute code after the user triggers a save but before the save actually occurs. Excel also provides an **AfterSave** event that can execute macro or add-in code after the save completes.
+
+When AutoSave is enabled, these events will fire automatically on a periodic basis without user interaction. Because of this new behavior when AutoSave is enabled, add-ins and macros that leverage these events may experience problems when AutoSave is on.
+
+In general, these issues can be avoided if the user chooses to disable AutoSave. You can also do this on the user’s behalf using the **autoSaveOn** property (see the example below). You can also take steps as a developer to mitigate these potential problems so that your add-ins and macros work smoothly, even if AutoSave is enabled.
+
+## Potential issues with save events and AutoSave
```
Comment (astegmaier) [inline VBA/Office-Shared-VBA/articles/about-autosave.md]: Unlike the other article, where each section only had 1-2 examples, the list here (5 examples) is sufficiently long that a summary table might be helpful. What do you think?

File: VBA/Excel-VBA/articles/application-workbookafterremotechange-event-excel.md (hunk 1 above)
Comment (ElizabethSamuel-MSFT) [inline VBA/Excel-VBA/articles/application-workbookafterremotechange-event-excel.md]: Yes, the sample code was tested and verified.

File: VBA/Excel-VBA/articles/about-coauthoring-in-excel.md [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+---
+
+
+# About co-authoring in Excel
+
+Learn about how co-authoring works in Excel 2016 for O365 subscribers and how you may need to adjust your add-in/macro for smooth integration with co-authoring.
+
+## About co-authoring
+
+Co-authoring enables you to edit a workbook hosted in the cloud (that is, OneDrive, OneDrive for Business, or SharePoint Online) simultaneously with other users. With each save, everyone editing the workbook at that time can see changes. If you're not ready for others to see your changes, then you can turn off [AutoSave](../../Office-Shared-VBA/articles/about-autosave.md) until you're ready to share your changes and receive others' changes.
```
Comment (ElizabethSamuel-MSFT) [inline VBA/Excel-VBA/articles/about-coauthoring-in-excel.md]: Updated.

File: VBA/Excel-VBA/articles/about-coauthoring-in-excel.md [hunk 10]
```diff
@@ -0,0 +1,118 @@
+---
+title: About co-authoring in Excel
+ms.prod: EXCEL
+ms.assetid: about-coauthoring-in-excel
+---
+
+
+# About co-authoring in Excel
+
+Learn about how co-authoring works in Excel 2016 for O365 subscribers and how you may need to adjust your add-in/macro for smooth integration with co-authoring.
```
Comment (ElizabethSamuel-MSFT) [inline VBA/Excel-VBA/articles/about-coauthoring-in-excel.md]: Updated.

File: VBA/Excel-VBA/articles/about-coauthoring-in-excel.md [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+
+## About co-authoring
+
+Co-authoring enables you to edit a workbook hosted in the cloud (that is, OneDrive, OneDrive for Business, or SharePoint Online) simultaneously with other users. With each save, everyone editing the workbook at that time can see changes. If you're not ready for others to see your changes, then you can turn off [AutoSave](../../Office-Shared-VBA/articles/about-autosave.md) until you're ready to share your changes and receive others' changes.
+
+**IMPORTANT**: Each instance of add-in or macro code runs independently and maintains its own internal state.
```
Comment (ElizabethSamuel-MSFT) [inline VBA/Excel-VBA/articles/about-coauthoring-in-excel.md]: Updated.

File: VBA/Excel-VBA/articles/about-coauthoring-in-excel.md [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+
+### Example scenario
```
Comment (ElizabethSamuel-MSFT) [inline VBA/Excel-VBA/articles/about-coauthoring-in-excel.md]: Updated.

File: VBA/Excel-VBA/articles/about-coauthoring-in-excel.md [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+
+Imagine an add-in that allows the user to create custom charts based on data in an Excel workbook. This add-in loads data for the user's charts into a hidden sheet in the workbook. When a file containing the custom charts is opened, the add-in reads data on the hidden sheet and loads the chart into memory. As the user makes edits to the chart, this in-memory structure is updated and re-written to the file before each save. This add-in assumes that the only time it is necessary to read the hidden sheet and load it into memory is when the file is opened.
+
+Co-authoring opens another possibility: the hidden sheet could be modified by another user running the same add-in at the same time. If this occurs, the charts that the users are viewing might become out of sync. For example:
+
+- Suppose User A opens the file and starts viewing an existing custom chart.
+- While she is doing this, User B opens the same file and starts making changes to the custom chart (for example, changes the zoom level of the chart).
```
Comment (ElizabethSamuel-MSFT) [inline VBA/Excel-VBA/articles/about-coauthoring-in-excel.md]: Removed.

File: VBA/Excel-VBA/articles/about-coauthoring-in-excel.md [hunk 14]
```diff
@@ -0,0 +1,118 @@
...
+- That change would be saved to the sheet by the add-in on User B’s computer, but User A would never see the change until she reloaded the file.
+
+### Workaround
+
+As much as possible try to avoid making assumptions about when workbook data can be changed. In this case, the developer could modify the add-in to react to the **AfterRemoteChange** event and check the hidden sheet’s values to see if they need to be read again by the add-in to allow User A to view the chart changes that User B made. The macro is intended to be run anytime the chart range is changed. This happens on load and can happen with a remote change. As such, your logic in **AfterRemoteChange** should re-run the macro.
```
Comment (ElizabethSamuel-MSFT) [inline VBA/Excel-VBA/articles/about-coauthoring-in-excel.md]: Linked to BeforeRemoteChange and AfterRemoteChange events where I listed them as new events.

File: VBA/Excel-VBA/articles/about-coauthoring-in-excel.md [hunk 15]
```diff
@@ -0,0 +1,118 @@
...
+
+## Integration with co-authoring events
+
+Co-authoring introduces new events **BeforeRemoteChange** and **AfterRemoteChange** which enable you to handle remote changes.
```
Comment (ElizabethSamuel-MSFT) [inline VBA/Excel-VBA/articles/about-coauthoring-in-excel.md]: Updated.

File: VBA/Excel-VBA/articles/about-coauthoring-in-excel.md [hunk 16]
```diff
@@ -0,0 +1,118 @@
...
+
+### When to receive remote changes
+
+You may want to receive the latest changes made by another user when the data is fundamental to the expected behavior of the workbook, for example, data visualization and the navigation task pane.
```
Comment (ElizabethSamuel-MSFT) [inline VBA/Excel-VBA/articles/about-coauthoring-in-excel.md]: Updated.

File: VBA/Excel-VBA/articles/about-coauthoring-in-excel.md [hunk 17]
```diff
@@ -0,0 +1,118 @@
...
+
+## Integration with co-authoring events
+
+Co-authoring introduces new events **BeforeRemoteChange** and **AfterRemoteChange** which enable you to handle remote changes.
+
+### When to receive remote changes
+
+You may want to receive the latest changes made by another user when the data is fundamental to the expected behavior of the workbook, for example, data visualization and the navigation task pane. 
+
+**Table 1. Examples where the user should receive remote changes**
```
Comment (ElizabethSamuel-MSFT) [inline VBA/Excel-VBA/articles/about-coauthoring-in-excel.md]: Removed.

File: VBA/Excel-VBA/articles/about-coauthoring-in-excel.md [hunk 18]
```diff
@@ -0,0 +1,118 @@
...
+
+|**Example**|**Scenario**|
+|:-----|:-----|
+|Data visualization|Your add-in plots data points on a map based on location data found in a particular range in the workbook. If a user edits any of the location data, all the remote users should receive this change so that each user's map can be updated.|
+|Navigation task pane|Your add-in displays all current workbook tabs in a task pane for easy navigation. If a user adds a worksheet, all the remote users should receive this change so that each user's task pane can display a link to the new worksheet.|
+
+### Data visualization example
+
+Let's say that you have created a custom map. In this example, you would add code to change location data then update the map. The workbook is shared with someone in a different city. With autosave on, the change is passed to the remote user who catches the change with the **AfterRemoteChange** event.
```
Comment (ElizabethSamuel-MSFT) [inline VBA/Excel-VBA/articles/about-coauthoring-in-excel.md]: Updated.

File: VBA/Excel-VBA/articles/about-coauthoring-in-excel.md [hunk 19]
```diff
@@ -0,0 +1,118 @@
...
+```vb
+Private Sub Workbook_AfterRemoteChange()
+    'Call subroutine that updates map
+End Sub
+```
+
+**Figure 1. Sample of London map with a few points of interest**
+![london locations](images/londonLocations.png) 
+
+### When to ignore remote changes
```
Comment (ElizabethSamuel-MSFT) [inline VBA/Excel-VBA/articles/about-coauthoring-in-excel.md]: Removed.

File: VBA/Excel-VBA/articles/about-coauthoring-in-excel.md [hunk 20]
```diff
@@ -0,0 +1,118 @@
...
+
+You may want to avoid changes that cause errors or degraded performance, for example, around data validation and data consistency. See Table 2 for example scenarios.
+
+**Table 2. Examples where the user should not react to remote changes**
+
+|**Example**|**Scenario**|
```
Comment (ElizabethSamuel-MSFT) [inline VBA/Excel-VBA/articles/about-coauthoring-in-excel.md]: Removed.

File: VBA/Excel-VBA/articles/about-coauthoring-in-excel.md [hunk 21]
```diff
@@ -0,0 +1,118 @@
...
+|:-----|:-----|
+|Data validation|A change event is triggered when a specific range is edited in the workbook. Your add-in code then validates the change and, if the check fails, notifies the user via pop-up dialog. However, if all the remote users collaborating on that workbook are notified about a validation failure unrelated to their own changes, this can lead to a poor experience.|
+|Data consistency|A change event is triggered and your add-in code synchronizes the data in the workbook with data in another part of the workbook or in an external system. If a remote user receives the change which causes the add-in code to synchronize the same data, this can lead to decreased performance for the remote user or data duplication in the external system.|
+
+### Data validation example
+
+For this example, we've created a chart that displays how much we've made selling various desserts. Neither the cost nor the number of items sold should be negative so there's a validation check that displays a message to the user.  When the invalid value is pushed to the remote users, the validation message should not be displayed to them.
```
Comment (ElizabethSamuel-MSFT) [inline VBA/Excel-VBA/articles/about-coauthoring-in-excel.md]: Updated.

File: VBA/Excel-VBA/articles/about-coauthoring-in-excel.md (hunk 13 above)
Comment (ElizabethSamuel-MSFT) [inline VBA/Excel-VBA/articles/about-coauthoring-in-excel.md]: Updated.

File: VBA/Office-Shared-VBA/articles/about-autosave.md (hunk 3 above)
Comment (ElizabethSamuel-MSFT) [inline VBA/Office-Shared-VBA/articles/about-autosave.md]: Updated.

File: VBA/Office-Shared-VBA/articles/about-autosave.md (hunk 2 above)
Comment (ElizabethSamuel-MSFT) [inline VBA/Office-Shared-VBA/articles/about-autosave.md]: Updated.

File: VBA/Office-Shared-VBA/articles/about-autosave.md (hunk 4 above)
Comment (ElizabethSamuel-MSFT) [inline VBA/Office-Shared-VBA/articles/about-autosave.md]: Updated.

File: VBA/Office-Shared-VBA/articles/about-autosave.md (hunk 5 above)
Comment (ElizabethSamuel-MSFT) [inline VBA/Office-Shared-VBA/articles/about-autosave.md]: Updated to "could be unresponsive"

File: VBA/Office-Shared-VBA/articles/about-autosave.md (hunk 6 above)
Comment (ElizabethSamuel-MSFT) [inline VBA/Office-Shared-VBA/articles/about-autosave.md]: Updated.

File: VBA/Office-Shared-VBA/articles/about-autosave.md (hunk 7 above)
Comment (ElizabethSamuel-MSFT) [inline VBA/Office-Shared-VBA/articles/about-autosave.md]: Updated.

File: VBA/Excel-VBA/articles/about-coauthoring-in-excel.md [hunk 22]
```diff
@@ -0,0 +1,118 @@
...
+
+- Suppose User A opens the file and starts viewing an existing custom chart.
+- While she is doing this, User B opens the same file and starts making changes to the custom chart (for example, changes the zoom level of the chart).
+- That change would be saved to the sheet by the add-in on User B’s computer, but User A would never see the change until she reloaded the file.
+
+### Workaround
+
+As much as possible try to avoid making assumptions about when workbook data can be changed. In this case, the developer could modify the add-in to react to the **AfterRemoteChange** event and check the hidden sheet’s values to see if they need to be read again by the add-in to allow User A to view the chart changes that User B made. The macro is intended to be run anytime the chart range is changed. This happens on load and can happen with a remote change. As such, your logic in **AfterRemoteChange** should re-run the macro.
+
+## Integration with co-authoring events
```
Comment (ElizabethSamuel-MSFT) [inline VBA/Excel-VBA/articles/about-coauthoring-in-excel.md]: Updated.

File: VBA/Excel-VBA/articles/about-coauthoring-in-excel.md [hunk 23]
```diff
@@ -0,0 +1,118 @@
...
+
+Co-authoring introduces new events **BeforeRemoteChange** and **AfterRemoteChange** which enable you to handle remote changes.
+
+### When to receive remote changes
```
Comment (ElizabethSamuel-MSFT) [inline VBA/Excel-VBA/articles/about-coauthoring-in-excel.md]: Updated.

File: VBA/Excel-VBA/articles/about-coauthoring-in-excel.md [hunk 24]
```diff
@@ -0,0 +1,118 @@
+---
+title: About co-authoring in Excel
```
Comment (ElizabethSamuel-MSFT) [inline VBA/Excel-VBA/articles/about-coauthoring-in-excel.md]: Updated.
